src/libcore/slice.rs

```diff
@@ -2177,6 +2177,16 @@ pub unsafe fn from_raw_parts_mut<'a, T>(p: *mut T, len: usize) -> &'a mut [T] {
 // Comparison traits
 //
 
+// A compile time optimization for faster bytewise slice comparison
+include!(concat!(env!("OUT_DIR"), "/memcmp_optimization.rs"));
+
+// The compare macro for the bytewise slice comparison
+macro_rules! cmp (
+    ($left:expr, $right: expr, $var:ident, $offset:expr) => {{
+        unsafe {*($left.offset($offset) as *const $var) == *($right.offset($offset) as *const $var)}
```

Comment:
This is a  possibly unaligned load (for example `*const u64` used that is not necessarily well aligned for `u64`. We can't merge the code like this, because it is not portable and will crash on certain platforms.)